### ISSUE TITLE
Optuna Tuning

### ISSUE THREAD

**Issue by AswinBalajiTR** [AswinBalajiTR/spring-2026-group6#13]
Define search space for each model
- Tune LSTM and CNN-LSTM hyperparameters
- Test different input/output window sizes
- Optimize learning rate, layers, hidden units, dropout, batch size, and epochs
- Compare best trials for each model
- Save best parameters and results
- Re-train final models using best settings
